src/infra/ssl/sslCmdRepo.go

```diff
@@ -173,75 +180,119 @@ func (repo *SslCmdRepo) shouldIncludeWww(vhost valueObject.Fqdn) bool {
-	if err != nil {
-		return errors.New("CreateOwnershipValidationMappingError: " + err.Error())
-	}
+			serverIpAddress = &ipAddress
+			break
+		}
 
-	isDomainMappedToServer := repo.isDomainMappedToServer(
-		firstVhostName,
-		expectedOwnershipHash,
-	)
+		if serverIpAddress == nil {
+			continue
+		}
 
-	mappingQueryRepo := mappingInfra.NewMappingQueryRepo(repo.persistentDbSvc)
-	mappings, err := mappingQueryRepo.ReadByHostname(firstVhostName)
-	if err != nil {
-		return errors.New("ReadVhostMappingsError: " + err.Error())
+		ownershipValidationMappingId, err := repo.createOwnershipValidationMapping(
+			vhostName,
+			expectedOwnershipHash,
+		)
+		if err != nil {
+			continue
+		}
+
+		hashUrlPath := infraData.GlobalConfigs.DomainOwnershipValidationUrlPath
+		hashUrlFull := "https://" + vhostNameStr + hashUrlPath
+		serverIpStr := serverIpAddress.String()
+
+		ownershipHashFound, err := infraHelper.RunCmd(
+			"curl", "-skLm", "10", "--resolve", vhostNameStr+":443:"+serverIpStr, hashUrlFull,
+		)
+		if err != nil {
+			hashUrlFull = "https://" + serverIpStr + hashUrlPath
+			ownershipHashFound, err = infraHelper.RunCmd(
+				"curl", "-skLm", "10", "-H", "Host: "+vhostNameStr, hashUrlFull,
+			)
+			if err != nil {
+				continue
+			}
+		}
+
+		if ownershipHashFound != expectedOwnershipHash.String() {
+			continue
+		}
+
+		domainsMappedToServer = append(domainsMappedToServer, vhostName)
+
+		if repo.shouldIncludeWww(vhostName) {
+			vhostNameWithWww, err := valueObject.NewFqdn("www." + vhostNameStr)
+			if err != nil {
+				continue
+			}
+
+			domainsMappedToServer = append(domainsMappedToServer, vhostNameWithWww)
+		}
+
+		mappingCmdRepo := mappingInfra.NewMappingCmdRepo(repo.persistentDbSvc)
+		err = mappingCmdRepo.Delete(ownershipValidationMappingId)
+		if err != nil {
+			log.Printf("DeleteOwnershipValidationMappingError: %s", err.Error())
+		}
 	}
 
-	if len(mappings) == 0 {
-		return errors.New("VhostMappingsNotFound")
+	return domainsMappedToServer
+}
+
+func (repo *SslCmdRepo) ReplaceWithValidSsl(sslPair entity.SslPair) error {
+	validPairVhostNames := repo.filterDomainsMappedToSomewhere(
+		sslPair.VirtualHostsHostnames,
+	)
+	if len(validPairVhostNames) == 0 {
+		return errors.New("NoDomainsMappedToSomewhere")
 	}
 
-	err = mappingCmdRepo.Delete(mappingId)
+	expectedOwnershipHash, err := repo.sslQueryRepo.GetOwnershipValidationHash(
+		sslPair.Certificate.CertificateContent,
+	)
 	if err != nil {
-		return errors.New("DeleteOwnershipValidationMappingError: " + err.Error())
+		return errors.New(
+			"CreateOwnershipValidationHashError: " + err.Error(),
+		)
 	}
-
-	if !isDomainMappedToServer {
-		return errors.New("DomainNotResolvingToServer")
+	pairVhostNamesMappedToServer := repo.filterDomainsMappedToServer(
+		validPairVhostNames, expectedOwnershipHash,
+	)
+	if len(pairVhostNamesMappedToServer) == 0 {
+		return errors.New("NoDomainsMappedToServer")
 	}
 
+	firstVhostName := pairVhostNamesMappedToServer[0]
```

Comment:
I think that the main domain of a SSL is called Common Name. You should use this term here to better identify the first ssl pair vhost as the common name.